### PR TITLE
[14.0][FW-PORT] helpdesk_mgmt: Email channel

### PR DESCRIPTION
**Pull request by ypapouin**
This PR has been split from #448

- Set "e-mail" channel when ticket generated from fetchmail cron task.

### CODE REVIEW

File: helpdesk_mgmt/tests/common.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def setUpClass(cls):
             context=ctx,
         )
         cls.stage_closed = cls.env.ref("helpdesk_mgmt.helpdesk_ticket_stage_done")
+        cls.channel_email = cls.env.ref("helpdesk_mgmt.helpdesk_ticket_channel_email")
```
Comment (pedrobaeza) [inline helpdesk_mgmt/tests/common.py]: No need to put this in the base. It's not adding too much value for the rest of the tests.

File: helpdesk_mgmt/models/helpdesk_ticket.py [hunk 2]
```diff
@@ -137,6 +137,14 @@ def create(self, vals_list):
...
             if vals.get("user_id") and not vals.get("assigned_date"):
                 vals["assigned_date"] = fields.Datetime.now()
+            # Automatically set e-mail channel when created from the
+            # fetchmail cron task
+            if self.env.context.get("fetchmail_cron_running") and not vals.get(
+                "channel_id"
+            ):
+                vals["channel_id"] = self.env.ref(
+                    "helpdesk_mgmt.helpdesk_ticket_channel_email"
+                ).id
```
Comment (pedrobaeza) [inline helpdesk_mgmt/models/helpdesk_ticket.py]: Protect this call with second argument `False` for avoiding an error if the channel is removed.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by PicchiSeba — COMMENTED** [review]
Code review: LGTM

**Comment by PicchiSeba** [comment]
I can't approve this PR, maybe because it's closed

**Pull request closed without merging**